File: src/java/org/javaturk/dp/ch05/proxy/basbakan/Vatandas.java
package org.javaturk.dp.ch05.proxy.basbakan;

public class Vatandas {
	private Basbakan basbakan;
	
	public Vatandas(BasbakanlikKalemi kalem){
		basbakan = kalem.banaBasbakaniVer();
	}

	public String derdiniAnlat(){
		basbakan.dertDinle("Bir derdim var...");
		return "Yasasiiinnn!!!";
	}
	
	public String isIste(){
		basbakan.isBul("Oglum");
		return "Yasasiiinnn!!!";
	}
}

File: src/c#/ch01/Vatandas.cs
using System;
namespace DesignPatterns.org.javaturk.dp.basic.pattern.gof.structural.proxy.basbakan
{
    public class Vatandas
    {
        private IBasbakan basbakan;

        public Vatandas(BasbakanlikKalemi kalem)
        {
            basbakan = kalem.BanaBasbakaniVer();
        }

        public String DerdiniAnlat()
        {
            basbakan.DertDinle("Bir derdim var...");
            return "Yasasiiinnn!!!";
        }


        public String IsIste()
        {
            basbakan.IsBul("Oglum");
            return "Yasasiiinnn!!!";
        }
    }
}
